python/ray/tests/test_label_scheduling.py

```diff
@@ -96,5 +96,92 @@ def test_label_selector_multiple(cluster_with_labeled_nodes):
     assert ray.get(actor.get_node_id.remote(), timeout=3) == node_3
 
 
+def test_fallback_strategy(cluster_with_labeled_nodes):
+    # Create a RayCluster with labelled nodes.
+    gpu_node, _, _ = cluster_with_labeled_nodes
+
+    # Define an unsatisfiable label selector.
+    infeasible_label_selector = {"ray.io/accelerator-type": "does-not-exist"}
+
+    # Create a fallback strategy with multiple accelerator options.
+    accelerator_fallbacks = [
+        {"label_selector": {"ray.io/accelerator-type": "A100"}},
+        {"label_selector": {"ray.io/accelerator-type": "TPU"}},
+    ]
+
+    # Attempt to schedule the actor. The scheduler should fail to find a node with the
+    # primary `label_selector` and fall back to the first available option, 'A100'.
+    label_selector_actor = MyActor.options(
+        label_selector=infeasible_label_selector,
+        fallback_strategy=accelerator_fallbacks,
+    ).remote()
+
+    # Assert that the actor was scheduled on the expected node.
+    assert ray.get(label_selector_actor.get_node_id.remote(), timeout=5) == gpu_node
+
+
+def test_empty_selector_fallback_strategy(cluster_with_labeled_nodes):
+    node_1, node_2, node_3 = cluster_with_labeled_nodes
+
+    # Define an unsatisfiable label selector.
+    infeasible_label_selector = {"ray.io/accelerator-type": "does-not-exist"}
+
+    # Create a fallback strategy with multiple label selector fallbacks. The
+    # first fallback option is unsatisfiable, so it falls back to the empty label
+    # selector option. This fallback should match any node.
+    accelerator_fallbacks = [
+        {"label_selector": {"ray.io/accelerator-type": "also-does-not-exist"}},
+        {"label_selector": {}},
+    ]
+
+    label_selector_actor = MyActor.options(
+        label_selector=infeasible_label_selector,
+        fallback_strategy=accelerator_fallbacks,
+    ).remote()
+
+    # Assert that the actor was scheduled on the expected node.
+    assert ray.get(label_selector_actor.get_node_id.remote(), timeout=5) in {
+        node_1,
+        node_2,
+        node_3,
+    }
```

Comment:
consider making the test more deterministic by scheduling 3 actors in parallel, each that occupies all CPUs on each node, and asserting that all 3 nodes are occupied by one of the actors